core/tests/performance_framework/perf_services.py

```diff
@@ -0,0 +1,227 @@
+        load timings for a page URL while simulating a cached session i.e, a
+        return user.
+        """
+        with Xvfb() as _:
+            self._setup_driver(use_proxy=False)
+
+            # Fetch the page once. This leads to caching of various resources.
+            self.driver.get(page_url)
+            self._wait_until_page_load_is_finished()
+
+            self.driver.get(page_url)
+            self._wait_until_page_load_is_finished()
+
+            page_session_timings = (
+                self.driver.execute_script("return window.performance"))
+
+        self._stop_driver()
+
+        return perf_domain.PageSessionMetrics(
+            page_session_stats=None, page_session_timings=page_session_timings)
+
+    def _wait_until_page_load_is_finished(self):
+        # Waits for the complete page to load, otherwise XHR requests
+        # made post initial page load will not be recorded.
+        time.sleep(self.WAIT_DURATION_SECS)
+
+    def _setup_proxy(self, downstream_kbps=None, upstream_kbps=None,
+                     latency=None):
+        self.server = Server(BROWSERMOB_PROXY_PATH)
+        self.server.start()
+        self.proxy = self.server.create_proxy()
+
+        # The proxy server is pretty sluggish, setting the limits might not
+        # achieve the desired behavior.
+        proxy_options = {}
+
+        if downstream_kbps:
+            proxy_options['downstream_kbps'] = downstream_kbps
+
+        if upstream_kbps:
+            proxy_options['upstream_kbps'] = upstream_kbps
+
+        if latency:
+            proxy_options['latency'] = latency
+
+        if len(proxy_options.items()) > 0:
+            self.proxy.limits(proxy_options)
+
+    def _setup_driver(self, use_proxy=False):
+        """Initializes a Selenium webdriver instance to programmatically
+        interact with a browser.
+        """
+        if self.browser == 'chrome':
+            chrome_options = webdriver.ChromeOptions()
+            # Disable several subsystems which run network requests in the
+            # background. This helps reduce noise when measuring network
+            # performance.
+            chrome_options.add_argument("--disable-background-networking")
+
+            if use_proxy:
+                proxy_url = urlparse.urlparse(self.proxy.proxy).path
+                proxy_argument = "--proxy-server={0}".format(proxy_url)
+                chrome_options.add_argument(proxy_argument)
+
+            self.driver = webdriver.Chrome(
+                CHROME_DRIVER_PATH,
+                chrome_options=chrome_options)
+
+        elif self.browser == 'firefox':
+            firefox_profile = webdriver.FirefoxProfile()
+
+            if use_proxy:
+                firefox_profile.set_proxy(self.proxy.selenium_proxy())
+
+            # pylint: disable=redefined-variable-type
+            self.driver = webdriver.Firefox(firefox_profile=firefox_profile)
+            # pylint: enable=redefined-variable-type
```

Comment:
also, to disable a line you can achieve by putting at the end of the line.

>  Is it possible to locally disable a particular message?
> Yes, this feature has been added in Pylint 0.11. This may be done by adding “#pylint: disable=some-message,another-one” at the desired block level or at the end of the desired line of code

source: https://docs.pylint.org/faq.html#message-control